src/System.Management.Automation/FormatAndOutput/common/Utilities/Mshexpression.cs

```diff
@@ -246,25 +301,36 @@ private MshExpressionResult GetValue(PSObject target, bool eatExceptions)
                     PSMemberInfo member = target.Properties[_stringValue];
                     if (member == null)
                     {
-                        return new MshExpressionResult(null, this, null);
+                        return new PSPropertyExpressionResult(null, this, null);
                     }
                     result = member.Value;
                 }
 
-                return new MshExpressionResult(result, this, null);
+                return new PSPropertyExpressionResult(result, this, null);
             }
             catch (RuntimeException e)
             {
                 if (eatExceptions)
                 {
-                    return new MshExpressionResult(null, this, e);
+                    return new PSPropertyExpressionResult(null, this, e);
                 }
                 else
                 {
                     throw;
                 }
             }
         }
+        
+        private PSObject IfHashtableWrapAsPSCustomObject(PSObject target)
```

Comment:
Can you please change the name to `WrapHashtableAsPSCustomObject`? C# method naming conversion suggests using verbs or verb phrases to name methods.

Reply:
The point of the rule is that the names of methods should be imperative which this is. There are other examples of (type 1) conditional clauses in .NET such as  `ThrowIfCancellationRequested` on `System.Threading.CancellationToken`.

Reply:
OK. #Closed